File: 881/881.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int l = 0, r = people.size() - 1;
        int result = 0;
        while (l <= r) {
            result++;
            if(people[l]+people[r]<=limit){
                l++;
            }
            r--;
        }
        return result;
    }
};

File: 881/881.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function (people, limit) {
  let l = 0,
    r = people.length - 1,
    result = 0;
  people.sort((a, b) => a - b);
  while (l < r) {
    if (people[l] + people[r] <= limit) l++;
    r--;
    result++;
  }
  if (l === r) result++;
  return result;
};
